components/zcash_note_encryption/src/lib.rs

```diff
@@ -1,8 +1,464 @@
-#[cfg(test)]
-mod tests {
-    #[allow(clippy::eq_op)]
-    #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
+//! Implementation of in-band secret distribution abstractions
+//! for Zcash transactions. The implementations here provide
+//! functionality that is shared between the Sapling and Orchard
+//! protocols.
+
+use crypto_api_chachapoly::{ChaCha20Ietf, ChachaPolyIetf};
+use rand_core::RngCore;
+
+pub const COMPACT_NOTE_SIZE: usize = 1 + // version
+    11 + // diversifier
+    8  + // value
+    32; // rcv
```

Comment:
```suggestion
    32; // rseed (or rcm prior to ZIP 212)
```

This should have been `rcm` and is now `rseed`.